code/hsec-core/src/Security/Advisories/Core/Advisory.hs

```diff
@@ -52,9 +55,24 @@ data Advisory = Advisory
   }
   deriving stock (Show)
 
-data ComponentIdentifier = Hackage Text | GHC GHCComponent
+data ComponentIdentifier
+  = Hackage PackageName
+  | Repository RepositoryURL RepositoryName PackageName
```

Comment:
I don't think you want to make Hackage more special than the others. Maybe a Pattern to build a Hackage package with pre-filled values would be useful, but I don't think it should be its own constructor.